OpenKeychain/src/main/java/org/sufficientlysecure/keychain/securitytoken/smartcardio/CommandAPDU.java

```diff
@@ -0,0 +1,605 @@
+/*
+ * Copyright (c) 2005, 2006, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package org.sufficientlysecure.keychain.securitytoken.smartcardio;
+
+import java.util.Arrays;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A command APDU following the structure defined in ISO/IEC 7816-4.
+ * It consists of a four byte header and a conditional body of variable length.
+ * This class does not attempt to verify that the APDU encodes a semantically
+ * valid command.
+ *
+ * <p>Note that when the expected length of the response APDU is specified
+ * in the {@linkplain #CommandAPDU(int,int,int,int,int) constructors},
+ * the actual length (Ne) must be specified, not its
+ * encoded form (Le). Similarly, {@linkplain #getNe} returns the actual
+ * value Ne. In other words, a value of 0 means "no data in the response APDU"
+ * rather than "maximum length."
+ *
+ * <p>This class supports both the short and extended forms of length
+ * encoding for Ne and Nc. However, note that not all terminals and Smart Cards
+ * are capable of accepting APDUs that use the extended form.
+ *
+ * <p>For the header bytes CLA, INS, P1, and P2 the Java type <code>int</code>
+ * is used to represent the 8 bit unsigned values. In the constructors, only
+ * the 8 lowest bits of the <code>int</code> value specified by the application
+ * are significant. The accessor methods always return the byte as an unsigned
+ * value between 0 and 255.
+ *
+ * <p>Instances of this class are immutable. Where data is passed in or out
+ * via byte arrays, defensive cloning is performed.
+ *
+ * @see ResponseAPDU
+ *
+ * @since   1.6
+ * @author  Andreas Sterbenz
+ * @author  JSR 268 Expert Group
+ */
+public final class CommandAPDU implements java.io.Serializable {
```

Comment:
and import java.io.Serializable instead of using the fully qualified package name